challenges/backend-challenge/src/main/java/com/number-submission/service/NumberSubmissionServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.numbersubmission.service;
+
+import java.util.List;
+
+@Service("numberSubmissionService")
+public class NumberSubmissionServiceImpl implements NumberService{
```

Comment:
I'd implement a caching mechanism, knowing that it's a safe assumption that the lookup value will not change often, so I'd save time and power with a cache. Thanks for this tip!

Reply:
Also, since the read is for the sorted array list, I may save and store the field as a sorted array in the cache.

Reply:
Yeah I think caching the sorted list (and potentially the unsorted list) would be the way to go here. You would slightly increase your memory usage but in this hypothetical scenario caching 500 ints is worth it 👍

Reply:
Are you speaking that 99% of the requests were calls to the Get request? I would likely just store the sorted array, instead of needing to call the sort each read. Is that what you were looking for with this question?